pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -1814,7 +1814,7 @@ func TestReconcile_SetsStartTime(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if err := testAssets.Controller.Reconciler.Reconcile(context.Background(), getRunName(taskRun)); err == nil {
+	if err := testAssets.Controller.Reconciler.Reconcile(testAssets.Ctx, getRunName(taskRun)); err == nil {
```

Comment:
A general rule of thumbs in theses tests : *never use `context.Background`* as `context.Context` for the `Reconcile` loop, as it bypass **all* the init magic `knative/pkg` does 😅

Reply:
So there is a little bit of magic this does, but the clients and informers are only injected on the `ctx` passed to `NewController` methods.  The `.Get()` methods aren't free (they are a list walk vs. field lookup), so we want folks to stash these during initialization and access them via fields.

So I'd say `context.Background()` is actually probably fine here, just not in calls to `NewController` :sweat`

Reply:
@mattmoor yeah, it's a bit tricky. The whole file would benefit from a small refactoring.